controller/monitor/disk_monitor.go

```diff
@@ -388,34 +388,34 @@ func NewDiskInfo(diskName, diskUUID, diskPath string, diskDriver longhorn.DiskDr
 	return diskInfo
 }
 
-func (m *NodeMonitor) getOrphanedReplicaInstanceNames(diskType longhorn.DiskType, diskUUID, diskPath string, replicaDirectoryNames map[string]string) (map[string]string, error) {
+func (m *NodeMonitor) getOrphanedReplicaDataStores(diskType longhorn.DiskType, diskUUID, diskPath string, replicaDataStores map[string]string) (map[string]string, error) {
 	switch diskType {
 	case longhorn.DiskTypeFilesystem:
-		return m.getOrphanedReplicaDirectoryNames(diskUUID, diskPath, replicaDirectoryNames)
+		return m.getOrphanedReplicaDirectoryNames(diskUUID, diskPath, replicaDataStores)
 	case longhorn.DiskTypeBlock:
-		return m.getOrphanedReplicaLvolNames(replicaDirectoryNames)
+		return m.getOrphanedReplicaLvolNames(replicaDataStores)
 	default:
 		return nil, fmt.Errorf("unknown disk type %v", diskType)
 	}
 }
 
-func (m *NodeMonitor) getOrphanedReplicaLvolNames(replicaDirectoryNames map[string]string) (map[string]string, error) {
-	if len(replicaDirectoryNames) == 0 {
+func (m *NodeMonitor) getOrphanedReplicaLvolNames(replicaDataStores map[string]string) (map[string]string, error) {
+	if len(replicaDataStores) == 0 {
 		return map[string]string{}, nil
 	}
 
-	for name := range replicaDirectoryNames {
+	for name := range replicaDataStores {
 		_, err := m.ds.GetReplica(name)
-		if err == nil || !datastore.ErrorIsNotFound(err) {
-			delete(replicaDirectoryNames, name)
+		if err == nil || !datastore.ErrorIsNotFound(err) || name == "" {
```

Comment:
Why will we have this case? `name` is empty?

Reply:
As reported in the issue description, somehow a v2 orphan `Spec.Parameters.DataName` is empty, I don't know why an empty string will be in the candidate list but I think at least this function should filter it out here. 
Without this change, getting a replica with an empty `name` will get `ErrorIsNotFound` then the empty string will be an orphaned data store, which shouldn't be.